src/inference/dev_api/openvino/runtime/icompiled_model.hpp

```diff
@@ -134,7 +134,7 @@ class OPENVINO_RUNTIME_API ICompiledModel : public std::enable_shared_from_this<
      *
      * @return OpenVINO RemoteContext
      */
-    std::shared_ptr<ov::IRemoteContext> get_context() const;
+    virtual std::shared_ptr<ov::IRemoteContext> get_context() const;
```

Comment:
have we tried using the meta context wrapper? do we still need virtual here?

Reply:
> meta context wrapper?

No didn't try context wrapper. I think add virtual will be more intuitive.

Reply:
Now I'm trying meta connext follow proxy plugin

Reply:
I think this change will avoid add viral in get_context(). 
https://github.com/openvinotoolkit/openvino/blob/ee06e0d0f4560fb75e2566be547c8279ce7f7aa8/src/inference/src/dev/icompiled_model.cpp#L155

Reply:
Verified! Auto batch can also using meta context. After this PR merged (https://github.com/openvinotoolkit/openvino/pull/18468) :
1) revert the change about get_context()
2) refactor auto batch remote context with meta context.
But anyway, pls continue the review, do u have another comment? @ilyachur @songbell @riverlijunjie